features/cellular/TARGET_UBLOX_MODEM/ublox_modem_driver/UbloxCellularInterface.h

```diff
@@ -0,0 +1,246 @@
+/* Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _UBLOX_CELLULAR_INTERFACE_
+#define _UBLOX_CELLULAR_INTERFACE_
+
+#include "mbed.h"
+#include "nsapi.h"
+#include "rtos.h"
+#include "FileHandle.h"
+#include "InterruptIn.h"
+#include "ATParser.h"
+
+// Forward declaration
+class NetworkStack;
```

Comment:
This implementation uses PPP to talk to modem and an external IP stack. So you need a forward declaration of NetworkStack. We will use external network stacks's socket operations, e.g., LWIP will provide those.